embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/utils/PropertyUtils.kt

```diff
@@ -22,47 +21,25 @@ internal object PropertyUtils {
      * @param properties properties to be normalized.
      * @return a normalized Map of the provided properties.
      */
-    @Suppress("UNCHECKED_CAST")
     @JvmStatic
     fun sanitizeProperties(properties: Map<String?, Any?>?): Map<String, Any> {
-        if (properties == null) {
-            return HashMap()
-        }
+        properties ?: return emptyMap()
```

Comment:
I would be wary of this change unless you've confirmed that none of the callers are assuming that this function always returns a `HashMap` and typecasting

Reply:
The only caller verifies that the `properties` is never null, so this case would not actually happen. I take another stab at cleaning this up further in a bit but will merge this as it is.